### PR TITLE
task: Rename interface{} to any

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Rename interface{} to any.

No need to review .go or .md files as they're auto-generated, review changes in templates and scripts.

NOTE: there are a few leftovers of interface{} in docs but don't seem easy to change using openapi-generator, maybe replacing strings in files.

Change tested in Terraform: https://github.com/mongodb/terraform-provider-mongodbatlas/compare/test_any?expand=1
Test execution: https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/11230665305

Link to any related issue(s): CLOUDP-276360

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: tools/scripts/generate.sh [hunk 1]
```diff
@@ -34,6 +34,7 @@ npm exec openapi-generator-cli -- generate \
     --package-name="$client_package" \
     --type-mappings=integer=int \
     --type-mappings=object=interface{} \
+    --type-mappings=interface{}=any \
```
Comment (lantoli) [inline tools/scripts/generate.sh:37]: important to do it in a line, if the change is:
```
--type-mappings=object=any \
```

then it'll create *any (pointer to any) for response params

File: tools/scripts/generate_mocks.sh [hunk 2]
```diff
@@ -7,3 +7,7 @@ if ! which mockery ; then
 else
 	mockery --dir ../mockadmin
 fi
+
+# Use any instead of interface{} in genereated mocks
```
Comment (lantoli) [inline tools/scripts/generate_mocks.sh:11]: update mocks to use any

**Review by EspenAlbert — APPROVED** [review]
LGTM

**Comment by oarbusi** [comment]
What's the reason for the change? does interface{} limit us in any way?

**Comment by wtrocki** [comment]
> No need to review .go or .md files as they're auto-generated, review changes in templates and scripts.

Some templates evaluate differently so it is good pattern to always check all source code.

> Change tested in Terraform

Nice. We do have examples folder we can also use for testing

**Comment by lantoli** [comment]
> What's the reason for the change? does interface{} limit us in any way?

@oarbusi they're the same, but any is more idiomatic for newer Go versions, I also took this opportunity to learn more about SDK.

**Pull request merged**